### ISSUE TITLE
RCS1256: does not recognize ArgumentNullException.ThrowIfNull

### ISSUE THREAD

**Issue by RichardRobertson** [JosefPihrt/Roslynator#1045]
**Product and Version Used**: 
Roslynator 4.2.0 with .Net 7

**Steps to Reproduce**:
Call [ArgumentNullException.ThrowIfNull](https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull) with a nullable argument.

**Actual Behavior**:
```csharp
void Method(object? obj)
{
    ArgumentNullException.ThrowIfNull(obj);
//  ^-------------------------------------^ RCS1256 
}
```

**Expected Behavior**:
```csharp
void Method(object? obj)
{
    ArgumentNullException.ThrowIfNull(obj);
//  no diagnostic reported
}
```

**Comment by ovska** [comment]
I think this behavior is intended. `obj` shouldn't be marked as nullable if you are throwing an exception every time.

From the [documentation](https://github.com/JosefPihrt/Roslynator/blob/main/docs/analyzers/RCS1256.md):
> This analyzer reports null checks of arguments that are:
> - **annotated as nullable reference type.**
> - optional and its default value is null.

**Comment by JosefPihrt** [comment]
@ovska is right.

**Comment by RichardRobertson** [comment]
Ok. I guess I misunderstood what the exception was telling me. I thought it meant what I did was not the right way to check for null. Maybe some wording should be added to the documentation indicating that the diagnostic actually means "I have a design issue and shouldn't be throwing on null in this context." I see that `if (obj == null)` and `if (obj is null)` also trigger RCS1256.

I'm building a wrapper for UIAutomationClient's COM object (because the managed library is woefully out of date and Microsoft keeps saying just use COM) and when it came to [AutomationIdentifier.CompareTo(object?)](https://learn.microsoft.com/en-us/dotnet/api/system.windows.automation.automationidentifier.compareto) I had this issue. The interface IComparable allows null to be passed to CompareTo. The design of the old managed library says that passing null results in an ArgumentNullException. I'm willing to accept that this abandoned library is wrong.

**Comment by GregReddick** [comment]
I have a class that is implementing ICustomFormatter. It requires that I implement the method 

> `public string Format(string? format, object? arg, IFormatProvider? formatProvider);`

However, in _my_ Format method, the `format `and `formatProvider` arguments must not be null and should throw an exception if they are. I cannot change the signature, and ThrowIfNull generates RCS1256.
